cookbook-1.md

```diff
@@ -10,67 +10,142 @@ As Service Consumer, call the API of a Service Provider without being pre-author
+function createClientAssertion(token) {
+  return new URLSearchParams({
     "grant_type": "client_credentials",
     "scope": "iSHARE",
-    "client_id": "EU.EORI.NLFLEXTRANS",
+    "client_id": YOUR_EORI,
     "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
-    "client_assertion": client_assertion
+    "client_assertion": token
+  })
+};
+
+// sign JWT payload with default settings
+function signJwt(payload) {
+  const header = {
+    alg: 'RS256',
+    typ: 'JWT',
+    x5c: x5c
+  };
+  return jwt.sign(payload, pemData, { algorithm: 'RS256', expiresIn: "30s", header: header });
 }
 
-response = requests.post(url, headers=headers, data=data)
+// Call /token endpoint and return access_token
+async function accessToken(eori, tokenUrl) {
+  let payload = { "iss": YOUR_EORI, "sub": YOUR_EORI, "aud": eori, "jti": uuidv4() }
+  const token = signJwt(payload);
+  response = await axios.post(tokenUrl, createClientAssertion(token), { "accept": "application/json", "Content-Type": "application/x-www-form-urlencoded" })
+  return response.data['access_token'];
+}
 
-print(response.status_code)
-print(response.json())
+// decode JWT without signature verification
+function decodeJWT(token) {
+  // Split the JWT into its three parts: header, payload, and signature
+  const parts = token.split('.');
+  if (parts.length !== 3) {
+      throw new Error('Invalid JWT');
+  }
+
+  // Decode the Base64Url encoded payload (second part)
+  const base64Url = parts[1];
+  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  const payload = decodeURIComponent(atob(base64).split('').map(function(c) {
+      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+  }).join(''));
+
+  // Parse the JSON payload
+  return JSON.parse(payload);
+}
 
-access_token = response.json()['access_token']
-```
+function checkAdherence(adh) {
+  if (adh['status'] !== 'Active') {
+    throw new Error("Status is not Active");
+  }
+  let now = new Date();
+  if (new Date(adh['start_date']) > now) {
+    throw new Error("Start date is set in future");
+  }
+  if (new Date(adh['end_date']) < now) {
+    throw new Error("End date is set in past");
+  }
+}
 
-### Call API with token
+let bearerToken = accessToken(ASSOC_EORI, tokenUrlAssoc);
+const headersParties = {
+  "accept": "application/json",
+  "Authorization": "Bearer " + bearerToken
+};
+
+const params = {
+  "active_only": "true",
+  "certified_only": "false",
+  "adherenceStatus": "Active",
+  "framework": "iSHARE",
+  "publiclyPublishable": "false",
+  "page": "1"
+};
+
+// association registry /parties
+response = await axios.get(partiesUrlAssoc + '/' + SP_EORI, { headers: headersParties, params: params })
+let partyToken = response.data['party_token'];
+
+const decodedPayload = decodeJWT(partyToken);
+console.log(decodedPayload);
+let party = decodedPayload["party_info"];
+console.log(party);
+checkAdherence(party);
+let ar = party["authregistery"][0];
+console.log(ar);
```

Comment:
Dit hele riedeltje is niet nodig als je geen call doet naar de authorization registry!